docs/tasks/LargeFileUploadTask.md

```diff
@@ -1,84 +1,142 @@
-# Large File Upload Task - Uploading large files to OneDrive
+# Large File Upload Task - Uploading large files to OneDrive, Outlook, Print API.
 
-This task simplifies the implementation of OneDrive's [resumable upload](https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/driveitem_createuploadsession).
+References - 
+- [Outlook's large file attachment](https://docs.microsoft.com/en-us/graph/outlook-large-attachments)
+- [OneDriveItem's resumable upload](https://docs.microsoft.com/en-us/graph/api/driveitem-createuploadsession?view=graph-rest-1.0&preserve-view=true)
+- [Print API's large file upload](https://docs.microsoft.com/en-us/graph/upload-data-to-upload-session)
 
 ## Creating the client instance
 
 Refer [this documentation](../CreatingClientInstance.md) for initializing the client.
 
-## Uploading from browser
 
-HTML to select the file for uploading.
-
-```HTML
-<input id="fileUpload" type="file" onchange="fileUpload(this)" />
-```
-
-Get files from the input element and start uploading.
+## Using the LargeFileUpload Task
 
+#### Create an upload session
+First step for any upload task is the creation of the upload session.
+**Example of a payload for Outlook**
 ```typescript
-async function fileUpload(elem) {
-	let file = elem.files[0];
-	try {
-		let response = await largeFileUpload(client, file, file.name);
-		console.log(response);
-		console.log("File Uploaded Successfully.!!");
-	} catch (error) {
-		console.error(error);
+const payload = {
+	AttachmentItem: {
+		attachmentType: 'file',
+		name: "<FILE_NAMe>",
+		size: FILE_SIZE,
 	}
 }
+```
+**Example of a payload for OneDrive**
+```typescript
+const payload = {
+			item: {
+				"@microsoft.graph.conflictBehavior": "rename",
+				name: "<FILE_NAMe>",
+		},
+	};
+```
+**Create the upload session**
+```typescript
+const uploadSession = LargeFileUploadTask.createUploadSession(client, "REQUEST_URL", payload)
+```
+#### Creating the LargeFileUploadTask object
 
-async function largeFileUpload(client, file) {
-	try {
-		let options = {
-			path: "/Documents",
-			fileName: file.name,
-			rangeSize: 1024 * 1024,
-		};
-		const uploadTask = await MicrosoftGraph.OneDriveLargeFileUploadTask.create(client, file, options);
-		const response = await uploadTask.upload();
-		return response;
-	} catch (err) {
-		throw err;
-	}
+* To create the LargeFileUploadTask object you need to create -
+    *  An upload session as shown above.
+    *  A `FileObject` instance.
+
+**FileObject Interface**
+```typescript
+export interface FileObject {
+	content: any;
```

Comment:
```suggestion
	content: ArrayBuffer | Blob | Buffer;
```

Reply:
documented the generic interface here. The point is to specify the generic nature of the content property enabling custom implementation as well.